[Yap Ho Wen] iP

howenyap opened the pull request:
# Snorlax
![snorlax](https://github.com/nus-cs2103-AY2324S2/ip/assets/108785851/5722c2fb-73ce-49b9-8f27-ee9fc9b54a5c)
>"Rest easy knowing your tasks are in good hands, at least I hope"

**Snorlax** helps you manage your tasks efficiently, it is:

- User-friendly
- Fast and reliable
- _Adorable_

## How do I Install it?
All you need to do is:

1. Download the `snorlax.jar` [here](https://github.com/howen02/ip/releases/download/Level-9/Snorlax.jar).
2. Open a command terminal, `cd` into the folder and type `java -jar snorlax.jar` to run
3. Start adding your tasks.
4. Let Snorlax handle the rest for you 💤

## Features
- [x] Task managing
- [ ] Automated task creation for reoccurring tasks 
- [ ] ~AI to analyse which task you should do first (probably never)~
---
If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(Duke.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

e0985925 committed:
Level-0

e0985925 committed:
Edits

e0985925 committed:
Abstracted line generation for easier symbol change

e0985925 committed:
Abstracted line generation for easier symbol change

e0985925 committed:
Echo feature added

e0985925 committed:
Task feature

e0985925 committed:
Abstracting command processing

e0985925 committed:
Modified stringlist into task and tasklist class

e0985925 committed:
added mark and unmark

e0985925 committed:
ToDos, Events, Deadlines

e0985925 committed:
Changes to marking tasks

e0985925 committed:
Changed Processor to use switch case

e0985925 committed:
Encapsulated greeting and exiting into processor

e0985925 committed:
Added file paths to FilePath class

e0985925 committed:
Automated file path editing when running testcases

e0985925 committed:
Automated Text UI Testing

e0985925 committed:
Rearranged files

e0985925 committed:
Added error handling for invalid command, invalid index and invalid deadline/event creation

e0985925 committed:
Moved printing statements from TaskList to new Command class

e0985925 committed:
added help command

e0985925 committed:
Fix imports

e0985925 committed:
Add EncaseLine, SingleLine, Greet and Exit to UI package

e0985925 committed:
Create utils package

e0985925 committed:
Edit utils

e0985925 committed:
Add storage

e0985925 committed:
Event, Deadline: Change String to LocalDateTime

e0985925 committed:
Commands package

e0985925 committed:
Enforce OOP

e0985925 committed:
Enforce more OOP

e0985925 committed:
Package objects

e0985925 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Merge as per insutructions

e0985925 committed:
Add gradle

e0985925 committed:
Add JUnit Tests

e0985925 committed:
Congifure shadowJar

e0985925 committed:
Write javadoc

e0985925 committed:
Enforce checkstyle format

e0985925 committed:
Add find function

yiwern5 commented on src/main/graphics/snorlax.txt:
Nice snorlax ASCII art 🤩

yiwern5 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,40 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        TaskList tasks = Storage.load();
```
Good use of plural nouns

yiwern5 commented on src/main/java/objects/Deadlines.java:
```diff
@@ -0,0 +1,35 @@
+     * @param name The name of the deadline task.
+     * @param by   The deadline date and time.
+     */
+    public Deadlines(String name, LocalDateTime by) {
```
Is it better to use a singular instead of plural noun for the class name? (same goes for ToDos and Events, considering Task is in singular)

yiwern5 commented on src/main/java/objects/Task.java:
```diff
@@ -0,0 +1,55 @@
+ */
+public class Task implements Serializable {
+    private final String name;
+    private boolean isDone;
```
Good job in correctly naming the boolean

yiwern5 commented on src/main/java/view/UI.java:
```diff
@@ -0,0 +1,13 @@
+/**
+ * UI is an abstract class providing a common structure for user interface classes.
+ */
+public abstract class UI {
```
Is it UI or Ui? 
![image](https://github.com/nus-cs2103-AY2324S2/ip/assets/111892633/4dfdc497-a4fb-49d3-95d8-6bd27a22aea5)

```suggestion
public abstract class Ui {
```

yiwern5 commented on src/main/java/view/CreatedTask.java:
```diff
@@ -0,0 +1,22 @@
+     * @param task  The newly created task.
+     */
+    public static void display(TaskList tasks, Task task) {
+        String o = String.format("Got it. I've added this task:\n  %s\nNow you have %d tasks in the list.",
```
Would it be better to replace the variable with a more meaningful name?

yiwern5 commented on src/test/java/commands/DeleteTaskTest.java:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    @Test
+    void execute_validIndex_taskRemoved() {
```
Good naming convention for test methods

yiwern5 reviewed (commented):
LGTM! Overall, your code is clean and easy to read. I've added some suggestions for you 🤩

e0985925 committed:
Follow styling guidelines

e0985925 committed:
JavaFX

e0985925 committed:
JavaFX

howen02 commented:
> LGTM! Overall, your code is clean and easy to read. I've added some suggestions for you 🤩

e0985925 committed:
Use assertions to document important assumptions in program

e0985925 committed:
Improve code quality through refractoring and adding more abstractions

howenyap committed:
Merge pull request #2 from howen02/branch-A-Assertions

Use assertions to document important assumptions in program

howenyap committed:
Merge pull request #3 from howen02/branch-A-CodeQuality

Improve code quality through refractoring and adding more abstractions

e0985925 committed:
Add snooze method

Snooze allows users to delay tasks by a default time of 1 ohur
Used polymorphism such that events/deadlines will be snoozed but not todos

e0985925 committed:
Merge snooze feature into master branch

e0985925 committed:
Add clear tasks feature
Users had to delete tasks one by one in previous version
Users can now clear all tasks by simply calling

e0985925 committed:
Edit code to follow git standard

e0985925 committed:
Change find and help command

e0985925 committed:
Change styling

e0985925 committed:
Add personality and more error handling

howenyap committed:
Update README.md

howenyap committed:
Update README.md

howenyap committed:
Create _configl.yml

e0985925 committed:
Change help text

e0985925 committed:
Merge branch 'master' of https://github.com/howen02/ip

Merge divergent branches

e0985925 committed:
Change Ui.png

howenyap committed:
Update README.md

e0985925 committed:
Catch error for non integer index

e0985925 committed:
Add easter egg

howenyap committed:
Make changes to adhere to feedback

howenyap committed:
Fix bugs from feedback

howenyap committed:
Fix bugs
